src/main.ts

```diff
@@ -0,0 +1,89 @@
+import "./styles/style.css";
+import { getData } from "./util";
+
+type responseType = { key: string; ascii: string; audio: string }[];
+
+const URL =
+  "https://raw.githubusercontent.com/nikhilsnayak/static-data-files/main/drum-kit-data.json";
+const MAX_COUNT = 125;
+
+const keysContainer = document.querySelector<HTMLDivElement>(".keys")!;
+const audiosContainer = document.querySelector<HTMLDivElement>(".audios")!;
+const keyTemplate = <HTMLTemplateElement>(
+  document.getElementById("key-template")!
+);
+const audioTemplate = <HTMLTemplateElement>(
+  document.getElementById("audio-template")!
+);
+
+function renderElements(data: responseType): void {
+  data.forEach((item) => {
+    const key = keyTemplate.content.cloneNode(true) as HTMLElement;
+    const audio = audioTemplate.content.cloneNode(true) as HTMLElement;
+
+    key.querySelector(".key")!.setAttribute("data-key", item.ascii);
+    key.querySelector("kbd")!.textContent = item.key;
+    key.querySelector(".sound")!.textContent = item.audio;
+    key.querySelector(".count")!.textContent = `0/${MAX_COUNT}`;
+    key
+      .querySelector<HTMLDivElement>(".progress-bar")!
+      .style.setProperty("--progress", "0%");
+
+    key.querySelector(".key")!.setAttribute("data-count", "0");
+
+    audio.querySelector("audio")!.src = `/audio/${item.audio}.wav`;
+    audio.querySelector("audio")!.setAttribute("data-key", item.ascii);
+
+    keysContainer.appendChild(key);
+    audiosContainer.appendChild(audio);
+  });
+}
+
+function configure() {
+  window.addEventListener("keydown", (e) => {
+    const audio = document.querySelector<HTMLAudioElement>(
+      `audio[data-key="${e.keyCode}"]`
+    )!;
+
+    const key = document.querySelector<HTMLDivElement>(
+      `.key[data-key="${e.keyCode}"]`
+    )!;
+
+    if (!audio) return;
+    audio.currentTime = 0;
+    audio.play();
+    key.classList.add("playing");
+    const count = parseInt(key.getAttribute("data-count")!);
+    key.setAttribute("data-count", (count + 1).toString());
+  });
+
+  const keys = document.querySelectorAll<HTMLDivElement>(".key");
+
+  keys.forEach((key) => {
+    key.addEventListener("transitionend", function (e: TransitionEvent) {
+      if (e.propertyName !== "transform") return;
+      this.classList.remove("playing");
+
+      const count = Math.min(
+        parseInt(this.getAttribute("data-count")!),
+        MAX_COUNT
+      );
+
+      this.querySelector(".count")!.textContent = `${count}/${MAX_COUNT}`;
+      this.querySelector<HTMLDivElement>(".progress-bar")!.style.setProperty(
```

Comment:
This will only update the frequency of current key hit. Remember when we change the frequency, the progress of all keys need to be updated as the total count keeps changing.